packages/aws-cdk-lib/aws-cloudfront-origins/lib/function-url-origin.ts

```diff
@@ -52,4 +78,84 @@ export class FunctionUrlOrigin extends cloudfront.OriginBase {
       originKeepaliveTimeout: this.props.keepaliveTimeout?.toSeconds(),
     };
   }
-}
\ No newline at end of file
+}
+
+/**
+ * An Origin for a Lambda Function URL with OAC.
+ */
+class FunctionUrlOriginWithOAC extends cloudfront.OriginBase {
+  private originAccessControl?: cloudfront.IOriginAccessControl;
+  private functionUrl: lambda.IFunctionUrl;
+  private readonly props: FunctionUrlOriginWithOACProps;
+
+  constructor(lambdaFunctionUrl: lambda.IFunctionUrl, props: FunctionUrlOriginWithOACProps = {}) {
+    const domainName = cdk.Fn.select(2, cdk.Fn.split('/', lambdaFunctionUrl.url));
+    super(domainName, props);
+    this.functionUrl = lambdaFunctionUrl;
+    this.originAccessControl = props?.originAccessControl;
+
+    this.props = props;
+
+    validateSecondsInRangeOrUndefined('readTimeout', 1, 180, props.readTimeout);
+    validateSecondsInRangeOrUndefined('keepaliveTimeout', 1, 180, props.keepaliveTimeout);
+  }
+
+  protected renderCustomOriginConfig(): cloudfront.CfnDistribution.CustomOriginConfigProperty | undefined {
+    return {
+      originSslProtocols: [cloudfront.OriginSslPolicy.TLS_V1_2],
+      originProtocolPolicy: cloudfront.OriginProtocolPolicy.HTTPS_ONLY,
+      originReadTimeout: this.props.readTimeout?.toSeconds(),
+      originKeepaliveTimeout: this.props.keepaliveTimeout?.toSeconds(),
+    };
+  }
+
+  public bind(scope: Construct, options: cloudfront.OriginBindOptions): cloudfront.OriginBindConfig {
+    const originBindConfig = super.bind(scope, options);
+
+    if (!this.originAccessControl) {
+      this.originAccessControl = new cloudfront.FunctionUrlOriginAccessControl(scope, 'FunctionUrlOriginAccessControl');
+    }
+
+    this.addInvokePermission(scope, options);
+
+    if (!this.validateAuthType()) {
+      cdk.Annotations.of(scope).addWarning(
```

Comment:
Confirmed that we should throw an error here instead of warning to enforce security best practice. Could you please update the error message as well to `The authType of the Function URL must be set to AWS_IAM when origin access control signing method is SIGV4_ALWAYS.`?

Reply:
This commit addresses the following changes:
https://github.com/aws/aws-cdk/pull/31339/commits/86f60d85c7d4f2ccc376aa45edb1c3e292901d32

- Modification from Warning to Error: This change updates the handling from a warning to an error, which corresponds to the comment in question.
- Detailed Escape Patch for cfnOriginAccessControl: This improves handling of unknown specifications, enhancing type safety.
- Error Handling within Validation Methods: Error handling is now managed within validation methods, reducing complex value passing between classes.